### DISCUSSION TITLE
Increase interval of results captures

### DISCUSSION THREAD

**Discussion by bjkiting** [johanmeijer/grott#397] [Ideas]
Hi All

Firstly thanks to all the devs that contributed to this great project! 

I recently managed to get everything up and running, I have grott running in a docker container which is communicating with  MQTT broker on my Home Assistant. 

One of the first things I tested was comparing the minute interval data set to that of the Growat API (obtained using the HA Growatt cloud API integration). This data has 10 min intervals. It is very interesting to see how it is not averaged values but appears to be instantaneous values at the start of each interval, rather than the more accurate average of the entire time frame.

Having this increased to minute intervals is already a massive improvement, as can be seen in the image below.

![WhatsApp Image 2023-08-23 at 07 58 39](https://github.com/johanmeijer/grott/assets/47182610/c061241a-5ae0-4fec-80ac-bd8d44d45c59)

Since then I tried to increase it even further by setting the update interval to 0.1 minutes on the ShineWifi-F router settings. This lead to an increased number of logs that looks like the following:
`- Growatt packet received:
		  <socket.socket fd=5, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('172.18.0.2', 5279), raddr=('192.168.1.20', 3840)>
	 - Data less then minimum record length, data not processed`
	
Is it possible to change any parameters such that the updating interval can be further increases?

**Comment by rany2** [comment]
> Is it possible to change any parameters such that the updating interval can be further increases?

Anything further will require you to use custom firmware for your ShineWifi stick.

**Reply by johanmeijer** [reply]
Do you run only grott or also grottserver?

De Growatt server can not keep up with low update intervals.  With grottserver you can achieve a better (lower)update interval. 

The message Data less than minimum record length are normal messages. This means that Grott will not process this message because it does not contain useful data (these are mostly "ping" records).

**Reply by bjkiting** [reply]
Thanks for this reply! I was only running Grott. I changed over to Grottserver and it has been running flawlessly!

**Comment by johanmeijer** [comment]
@bjkiting can we close this one?

**Comment by SebastianGode** [comment]
@johanmeijer @bjkiting 
How did you change the update intervall to 0.1minute? I can only enter a full number there? Or am I too dumb to do that?
Also what happens if I change it down to 0.1minutes and don't use grottserver. Will the official Growatt server simply block me or not process the data? Grott should update it on MQTT fast without issues I assume.

**Reply by bjkiting** [reply]
As far as I know there are two ways to set it to less than a minute. The first (this is what I did to get the graph above) is to set it to 0.1 on the web interface. The second way is to use grottserver and change the registry value of the datalogger. As far as what the official Growatt server will do,  I can tell you that my setup is currently on 0.1 min and all the data is still available on the official Growatt server. I only use grottserver when I want to change the registries.

**Reply by SebastianGode** [reply]
How can you set it to 0.1 on the web interface?
Can you do some screenshots on when you are able to access that web interface and which IP?
I'd really really like to do this! @bjkiting

**Reply by SebastianGode** [reply]
@bjkiting Okay I set it to 0.1 in the Web-UI of server.growatt.com as my Stick firmware doesn't have a local web interface anymore.
What I noticed however is pretty interesting. When set to 1 it will push the useful message to grott every minute, exactly every minute on the second exakt. When setting it to 0.1 it isn't every minute anymore but roughly every 1min10s for me. Can you verify that? Now what may happen if we add a negative value like -0.1? Will it shrink to 50s?

**Reply by PedroKTFC** [reply]
Can you provide a detailed description of how you set in the web UI please? I had to use the hotspot method which is a right pain (especially as my datalogger is in the loft!).

**Comment by johanmeijer** [comment]
Be aware if you do not use grottserver changing the interval time might give strange values while the Growatt server does not really like short intervals (lower then 5 minutes). They have to serve a lot of inverters sending the data. This will produce a lot more of data they have to process.
